app/models/manageiq/providers/redhat/infra_manager/refresh_parser.rb

```diff
@@ -334,9 +340,28 @@ def self.host_inv_to_network_hashes(inv, guest_device_uids)
   def self.host_inv_to_os_hash(inv, hostname)
     return nil if inv.nil?
 
-    result = {:name => hostname}
-    result[:product_name] = 'linux'
-    result
+    {
+      :name         => hostname,
+      :product_type => 'linux',
+      :product_name => extract_host_os_name(inv),
+      :version      => extract_host_os_full_version(inv[:os])
+    }
+  end
+
+  def self.extract_host_os_full_version(host_os)
```

Comment:
@blomquisg There is a `nil` check in `host_inv_to_os_hash` as well - see untouched line 341 from the snippet above:

`def self.host_inv_to_os_hash(inv, hostname)
    return nil if inv.nil?
    ...
`